src/SAML2/SOAPClient.php

```diff
@@ -88,6 +88,10 @@ public function send(Message $msg, SimpleSAML_Configuration $srcMetadata, Simple
             $ctxOpts['ssl']['cafile'] = $peerCertFile;
         }
 
+        if ($srcMetadata->hasValue('saml.SOAPClient.stream_context.ssl.peer_name')) {
```

Comment:
Add the ability to change the SSL peer_name stream_context. This is a workaround for [a bug](http://php.net/manual/en/migration56.openssl.php) where peer verification fails when using a proxy with SoapClient (ext/soap). The bug is present since [PHP 5.6.x where they updated the underlying openssl library](http://php.net/manual/en/migration56.openssl.php).

The fix has a low impact to the SAML2 library and as configuring a stream_context is a perfectly normal feature for any stream application it does no harm to the scope of the SAML2 library.